bigtable/admin.go

```diff
@@ -2175,3 +2184,250 @@ func (ac *AdminClient) UpdateBackup(ctx context.Context, cluster, backup string,
 	_, err := ac.tClient.UpdateBackup(ctx, req)
 	return err
 }
+
+// AuthorizedViewConf contains information about an authorized view.
+type AuthorizedViewConf struct {
+	TableID          string
+	AuthorizedViewID string
+
+	// Types that are valid to be assigned to AuthorizedView:
+	//  *SubsetViewConf
+	AuthorizedView     isAuthorizedView
+	DeletionProtection DeletionProtection
+}
+
+// A private interface that currently only implemented by SubsetViewConf, ensuring that only SubsetViewConf instances are accepted as an AuthorizedView.
+// In the future if a new type of AuthorizedView is introduced, it should also implements this interface.
+type isAuthorizedView interface {
+	isAuthorizedView()
+}
+
+func (*SubsetViewConf) isAuthorizedView() {}
+
+// SubsetViewConf contains configuration specific to an authorized view of subset view type.
+type SubsetViewConf struct {
+	RowPrefixes   [][]byte
+	FamilySubsets map[string]FamilySubset
+}
+
+func (av AuthorizedViewConf) proto() *btapb.AuthorizedView {
+	var avp btapb.AuthorizedView
+
+	switch dp := av.DeletionProtection; dp {
+	case Protected:
+		avp.DeletionProtection = true
+	case Unprotected:
+		avp.DeletionProtection = false
+	default:
+		break
+	}
+
+	switch avt := av.AuthorizedView.(type) {
+	case *SubsetViewConf:
+		avp.AuthorizedView = &btapb.AuthorizedView_SubsetView_{
+			SubsetView: avt.proto(),
+		}
+	default:
+		break
+	}
+	return &avp
+}
+
+// GetSubsetView returns nil if the type of the AuthorizedView is not SubsetViewConf, otherwise it returns the pointer to the SubsetViewConf instance.
+func (av *AuthorizedViewConf) GetSubsetView() *SubsetViewConf {
+	if sv, ok := av.AuthorizedView.(*SubsetViewConf); ok {
+		return sv
+	}
+	return nil
+}
```

Comment:
Great point about returning (*Info) types for the Get method. Done.

The `GetSubsetView()` is meant to be a convenient method so users do not need to cast the interface. I don't think it makes a lot sense to have a function to return the un-casted interface, since the users can directly access the `AuthorizedView` field in the struct.

Reply:
If we follow this path, whenever we end up adding another kind of AuthorizedView we will have to add another getter for that type. Then we will have 2 getters that share the same storage. When looking at the api, the shared storage will be non-obvious. In the case of end users doing the cast themselves, it becomes obvious that only one type of AuthorizedView can be set